applications/ConstitutiveLawsApplication/custom_constitutive/auxiliary_files/hcf_data_container.cpp

```diff
@@ -0,0 +1,239 @@
+{
+    rFatigueVariables.MaxStress = mMaxStress;
+    rFatigueVariables.MinStress = mMinStress;
+    rFatigueVariables.MaxIndicator = mMaxDetected;
+    rFatigueVariables.MinIndicator = mMinDetected;
+    rFatigueVariables.PreviousStresses = mPreviousStresses;
+    rFatigueVariables.FatigueReductionFactor = mFatigueReductionFactor;
+    rFatigueVariables.ReversionFactorRelativeError = mReversionFactorRelativeError;
+    rFatigueVariables.MaxStressRelativeError = mMaxStressRelativeError;
+    rFatigueVariables.GlobalNumberOfCycles = mNumberOfCyclesGlobal;
+    rFatigueVariables.LocalNumberOfCycles = mNumberOfCyclesLocal;
+    rFatigueVariables.PreviousMaxStress = mPreviousMaxStress;
+    rFatigueVariables.PreviousMinStress = mPreviousMinStress;
+    rFatigueVariables.WohlerStress = mWohlerStress;
+    rFatigueVariables.NewCycle = false;
+    rFatigueVariables.Sth = mThresholdStress;
+    rFatigueVariables.CyclesToFailure = mCyclesToFailure;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void HCFDataContainer::UpdateFatigueVariables(HCFDataContainer::FatigueVariables &rFatigueVariables)
+{
+    mMaxStress = rFatigueVariables.MaxStress;
+    mMinStress = rFatigueVariables.MinStress;
+    mMaxDetected = rFatigueVariables.MaxIndicator;
+    mMinDetected = rFatigueVariables.MinIndicator;
+    mNumberOfCyclesGlobal = rFatigueVariables.GlobalNumberOfCycles;
+    mNumberOfCyclesLocal = rFatigueVariables.LocalNumberOfCycles;
+    mNewCycleIndicator = rFatigueVariables.NewCycle;
+    mPreviousMaxStress = rFatigueVariables.PreviousMaxStress;
+    mPreviousMinStress = rFatigueVariables.PreviousMinStress;
+    mPreviousStresses = rFatigueVariables.PreviousStresses;
+    mFatigueReductionFactor = rFatigueVariables.FatigueReductionFactor;
+    mWohlerStress = rFatigueVariables.WohlerStress;
+    mThresholdStress = rFatigueVariables.Sth;
+    mReversionFactorRelativeError = rFatigueVariables.ReversionFactorRelativeError;
+    mMaxStressRelativeError = rFatigueVariables.MaxStressRelativeError;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void HCFDataContainer::FinalizeSolutionStep(HCFDataContainer::FatigueVariables &rFatigueVariables,
+                        const Properties& rMaterialProperties,
+                        const ProcessInfo& rCurrentProcessInfo,
+                        ConstitutiveLaw::StressVectorType stress_vector,
+                        double uniaxial_stress)
+{
+    double sign_factor = CalculateTensionOrCompressionIdentifier(stress_vector);
+    uniaxial_stress *= sign_factor;
+
+    CalculateSminAndSmax(uniaxial_stress, rFatigueVariables);
+
+    rFatigueVariables.AdvanceStrategyApplied = rCurrentProcessInfo[ADVANCE_STRATEGY_APPLIED];
```

Comment:
retrieving something without checking if has been set is dangerous... Think that somebody could use this class without the AITS. I'd use: 

``
```suggestion
    rFatigueVariables.AdvanceStrategyApplied = rCurrentProcessInfo.Has(ADVANCE_STRATEGY_APPLIED) ? rCurrentProcessInfo[ADVANCE_STRATEGY_APPLIED : false;
```